cmd/collector/app/span_processor.go

```diff
@@ -177,3 +209,79 @@ func (sp *spanProcessor) enqueueSpan(span *model.Span, originalFormat SpanFormat
 	}
 	return sp.queue.Produce(item)
 }
+
+func (sp *spanProcessor) StartStateReporting(reportPeriod time.Duration) {
+	sp.background(reportPeriod, sp.updateGauges)
+}
+
+func (sp *spanProcessor) StartDynQueueSizeUpdater(reportPeriod time.Duration) {
+	sp.background(reportPeriod, sp.updateQueueSize)
+}
```

Comment:
Why not just call `sp.background(reportPeriod, sp.updateQueueSize)` directly? These two functions are only called once each.

Why do they need to be public?

Reply:
+1, I think this code was bigger in a previous iteration. In its current state, doesn't make sense to have it as a separate function.